add `--strict-vars` option to the `wrangler types` command (#18981)
cloudflare/cloudflare-docs: src/content/docs/workers/wrangler/commands.mdx

  - Leave the path blank to use the default option, e.g. `npx wrangler types --x-include-runtime`
  - A custom path must be relative to the project root, e.g. `./my-runtime-types.d.ts`
  - A custom path must have a `d.ts` extension.
- `--loose-vars` <Type text="boolean" /> <MetaInfo text="optional (default: false)" />

>   - Flag used to generate "loose"/generic types instead of literal and union types for variables (e.g. `myEnv: string` instead of `myEnv: 'my env variable'`)

Author: we've got `e.g.` all over the place

  - If `true`, (the default) Wrangler generates literal and union types for bindings (e.g. `myEnv: 'my dev variable' | 'my prod variable'`).

>   - If `false`, Wrangler generates generic types (e.g. `myEnv: string`) (useful when environment variables can often change especially when multiple environments are used).

Reviewer: ```suggestion
  - If `false`, Wrangler generates generic types (e.g. `myEnv: string`). This is useful when variables change frequently, such as when working across multiple environments.
```

  - Leave the path blank to use the default option, e.g. `npx wrangler types --x-include-runtime`
  - A custom path must be relative to the project root, e.g. `./my-runtime-types.d.ts`
  - A custom path must have a `d.ts` extension.
- `--strict-vars` <Type text="boolean" /> <MetaInfo text="optional (default: true)" />
  - Control the types that Wrangler generates for `vars` bindings.

>   - If `true` (the default) Wrangler generates literal and union types for bindings (e.g. `myEnv: 'my dev variable' | 'my prod variable'`).

Reviewer: ```suggestion
  - If `true`, (the default) Wrangler generates literal and union types for bindings (e.g. `myEnv: 'my dev variable' | 'my prod variable'`).
```

  - If `false`, Wrangler generates generic types (e.g. `myEnv: string`).

>   - Useful when environment variables can often change (especially with multiple environments are involved)

Reviewer: ```suggestion
  - Useful when environment variables can often change (especially with multiple environments are involved).
```

Reviewer: Maybe this would be clear to the reader, but it wasn't clear to me if this bullet-point is referring to the entire command or just false. If it's referring to the entire command, I would just make it the second sentence of the first bullet point, which explains what it is and when to use it.

Author: yeah I see what you mean, I totally agree that this might be confusing 👍 

It refers to only the `false` case, so maybe this and the bullet point above should be merged?

Reviewer: Yup! Then I'd just add it as a second second to that last bullet :)

  - Leave the path blank to use the default option, e.g. `npx wrangler types --x-include-runtime`
  - A custom path must be relative to the project root, e.g. `./my-runtime-types.d.ts`
  - A custom path must have a `d.ts` extension.
- `--strict-vars` <Type text="boolean" /> <MetaInfo text="optional (default: true)" />

>   - Flag to opt out of the default generation of literal and union types for variables in favour of more generic types (e.g. to generate `myEnv: string` instead of `myEnv: 'my env variable'`)

Reviewer 2: How about more explicitly:

```suggestion
  - Control the types that Wrangler generates for `vars` bindings.
  - If true (the default) Wrangler generates literal and union types for bindings (e.g. `myEnv: 'my dev variable' | `my prod variable`).
  - If false, Wrangler generates generic types (e.g. `myEnv: string`).
```

Author: sounds good to me 🙂 

although true and false should be in backticks right?
